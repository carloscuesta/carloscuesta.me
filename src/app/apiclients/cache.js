const nodeCache = require('memory-cache')
const CACHE_TIME = 86400000

class CacheApiClient {
  serialize (params) {
    return Object.keys(params).map((key) => {
      return key + '=' + params[key]
    }).join('&')
  }

  validate (src, params, time) {
    const _self = this
    const url = `${_self.base_url}${src}?${this.serialize(params)}`
    const cacheTime = time * 6000 || CACHE_TIME
    const dataCache = nodeCache.get(url)

    if (!dataCache) {
      return _self.get(src, params).then((data) => {
        nodeCache.put(url, data, cacheTime)
        return data
      })
    } else {
      return Promise.resolve(dataCache)
    }
  }
}

module.exports = new CacheApiClient()
