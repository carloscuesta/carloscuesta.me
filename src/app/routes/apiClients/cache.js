const nodeCache = require('memory-cache')
const CONFIG = require('./config')

class CacheApiClient {
  serialize (params) {
    return Object.keys(params).map((key) => {
      return key + '=' + params[key]
    }).join('&')
  }

  validate (src, params, time) {
    const _self = this
    const url = `${_self.base_url}${src}?${this.serialize(params)}`
    const cacheTime = time * 6000 || CONFIG.CACHE_TIME
    const dataCache = nodeCache.get(url)

    if (!dataCache) {
      return _self.get(src, params).then((data) => {
        nodeCache.put(url, data, cacheTime)
        return data
      })
    } else {
      return Promise.resolve(dataCache)
    }
  }
}

module.exports = new CacheApiClient()
