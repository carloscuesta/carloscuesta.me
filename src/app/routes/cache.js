const cache = require('memory-cache')

const cacheClean = (req, res) => {
  cache.clear()
  res.redirect('/')
}

module.exports = cacheClean
