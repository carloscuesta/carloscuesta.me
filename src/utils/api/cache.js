// @flow
import NodeCache from 'node-cache'

const apiCache: { set: Function, get: Function } = new NodeCache({ stdTTL: 7200 })

export default apiCache
