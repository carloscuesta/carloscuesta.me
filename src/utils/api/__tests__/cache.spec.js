import cache from '../cache'

describe('api cache', () => {
  it('should match apiCache', () => {
    expect(cache).toMatchSnapshot()
  })
})
