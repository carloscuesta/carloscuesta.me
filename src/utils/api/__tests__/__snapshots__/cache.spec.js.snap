// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api cache should match apiCache 1`] = `
NodeCache {
  "ERRORS": Object {
    "ECACHEFULL": [Function],
    "EKEYSTYPE": [Function],
    "EKEYTYPE": [Function],
    "ENOTFOUND": [Function],
    "ETTLTYPE": [Function],
  },
  "_check": [Function],
  "_checkData": [Function],
  "_error": [Function],
  "_events": Object {},
  "_eventsCount": 0,
  "_getValLength": [Function],
  "_initErrors": [Function],
  "_isInvalidKey": [Function],
  "_maxListeners": undefined,
  "_wrap": [Function],
  "checkTimeout": 2,
  "close": [Function],
  "data": Object {},
  "del": [Function],
  "flushAll": [Function],
  "flushStats": [Function],
  "get": [Function],
  "getStats": [Function],
  "getTtl": [Function],
  "has": [Function],
  "keys": [Function],
  "mget": [Function],
  "mset": [Function],
  "options": Object {
    "arrayValueSize": 40,
    "checkperiod": 600,
    "deleteOnExpire": true,
    "enableLegacyCallbacks": false,
    "forceString": false,
    "maxKeys": -1,
    "objectValueSize": 80,
    "promiseValueSize": 80,
    "stdTTL": 7200,
    "useClones": true,
  },
  "set": [Function],
  "stats": Object {
    "hits": 0,
    "keys": 0,
    "ksize": 0,
    "misses": 0,
    "vsize": 0,
  },
  "take": [Function],
  "ttl": [Function],
  "validKeyTypes": Array [
    "string",
    "number",
  ],
}
`;
